### PR TITLE
Update tuner docs

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

Missed one usecase for batch size finder where it can be used with validate/test/predict as well.

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @rohitgr7 @Felonious-Spellfire @akihironitta

### CODE REVIEW

**Review by awaelchli — APPROVED** [review]

**Review by krshrimali — APPROVED** [review]

File: docs/source-pytorch/advanced/training_tricks.rst [hunk 1]
```diff
@@ -191,29 +191,52 @@ The algorithm in short works by:
...
+            super().__init__(*args, **kwargs)
+
+        def on_fit_start(self, *args, **kwargs):
+            return
+
+        def on_test_start(self, trainer, pl_module):
+            self.scale_batch_size(trainer, pl_module)
+
+
+    trainer = Trainer(callbacks=[EvalBatchSizeFinder()])
```
Comment (carmocca) [inline docs/source-pytorch/advanced/training_tricks.rst:238]: I thought this wasn't necessary and instead you would pass `method="test"`

Comment (rohitgr7) [reply docs/source-pytorch/advanced/training_tricks.rst:238]: that is for `trainer.tune(... , method='test')`

Comment (rohitgr7) [reply docs/source-pytorch/advanced/training_tricks.rst:238]: can be a little confusing.. let's discuss the `tune` API future after release.

Comment (carmocca) [reply docs/source-pytorch/advanced/training_tricks.rst:238]: My question is more along the lines of "When would you want to subclass vs using `tune(..., method="test")`"?

Are you saying we would be removing `.tune()` which would make this snippet the recommended approach?

Comment (rohitgr7) [reply docs/source-pytorch/advanced/training_tricks.rst:238]: for validate/test/predict, this would be same.. i.e using `trainer.tune(method='test')` would do the job.. 
but now since we have 2 ways to do the same thing, I was thinking if we could remove `trainer.tune` altogether since callbacks can do the job now. The reason I added callback in the example instead of tune is so that in case we move forward with removing it in the future, users should start using callbacks from now since we are not going to remove these callbacks.

Comment (rohitgr7) [reply docs/source-pytorch/advanced/training_tricks.rst:238]: if we keep `trainer.tune, I'll update the docs adding this API approach.

Comment (carmocca) [reply docs/source-pytorch/advanced/training_tricks.rst:238]: Thanks for the clarification! Removing tune makes sense to me but that should be discussed in a separate issue.

Comment (rohitgr7) [reply docs/source-pytorch/advanced/training_tricks.rst:238]: yeap!
